tools/convert-old-video-settings-file.py

```diff
@@ -0,0 +1,156 @@
+        """Update class internal configuration about video source and resolution
+
+        Args:
+            pair (tuple): Tuple that contains (key, value) for pipeline
+            settings dict
+        """
+
+        (key, value) = pair
+        self.pipeline_settings[key] = value
+
+    @staticmethod
+    def create_source_pipeline(device='/dev/video0', width=1920, height=1080, framerate=30):
+        """Populate and generate gstreamer v4l2src pipeline
+
+        Args:
+            device (str, optional): Video device. Defaults to '/dev/video0'.
+            width (int, optional): Video width. Defaults to 1920.
+            height (int, optional): Video height. Defaults to 1080.
+            framerate (int, optional): Video framerate. Defaults to 30.
+
+        Returns:
+            str: Return string with the pipeline
+        """
+
+        return Convert.SOURCE_PIPELINE.format(
+            device=device,
+            width=width,
+            height=height,
+            framerate=framerate
+        )
+
+    @staticmethod
+    def get_content_trash_file(filepath: str):
+        """Get content from file and move it to the trash
+
+        Args:
+            filepath (str): Path for the parameter file
+
+        Returns:
+            str: File content
+        """
+
+        if not os.path.isfile(filepath):
+            print("File: '{}' does not exist.".format(filepath))
+            return None
+
+        file_object = open(filepath, 'r')
+        content = file_object.read()
+        file_object.close()
+
+        filename = os.path.basename(filepath)
+        # TODO: Remove this before merge
+        #os.rename(filepath, os.path.join(Convert.TRASH, filename))
+
+        return content
+
+    def run(self):
+        """Run and extract pipeline from parameter files
+
+        Returns:
+            str: Final gstreamer pipeline
+        """
+
+        # Create trash folder if does not exist
+        try:
+            os.makedirs(self.TRASH)
+        except FileExistsError:
+            pass
+
+        # Get files content
+        vidformat_file_content = self.get_content_trash_file(
+            self.vidformat_file)
+        gstreamer2_file_content = self.get_content_trash_file(
+            self.gstreamer2_file)
+
+        # Get content of each setting file
+        # Extract the necessary information to build our new pipeline
+        if vidformat_file_content:
+            print(vidformat_file_content)
+            lines = vidformat_file_content.split('\n')
+            for name, value in zip(['width', 'height', 'framerate'], lines):
+                self.update_configuration((name, int(value)))
+
+        user_parse_pipeline = ''
+        if gstreamer2_file_content:
+            # Break each line as a pipeline element
+            # And join all elements to create a pipeline
+            lines = gstreamer2_file_content.split('\n')
+            lines = [line.replace('!', '').strip() for line in lines]
```

Comment:
this will fail if users have more than one element at the same line.

Reply:
Good catch